src/lib/validation.ts

```diff
@@ -0,0 +1,432 @@
+		case 'false':
+		case false:
+		case 'off':
+		case 0:
+			return false;
+		default:
+			return;
+	}
+}
+
+/*
+ * isValidShortText
+ *
+ * Check that the input string is definitely a string,
+ * and has a length which is less than 255
+ */
+export function isValidShortText(t: string): boolean {
+	return _.isString(t) && t.length <= 255;
+}
+
+/**
+ * isValidEnv
+ *
+ * Given a env var object, check types and values for the keys
+ * and values
+ */
+export function isValidEnv(obj: EnvVarObject): boolean {
+	if (!_.isObject(obj)) {
+		console.log('debug: Non-object passed to validation.isValidEnv');
+		console.log(`\tobj: ${inspect(obj)}`);
+		return false;
+	}
+
+	return _.every(obj, (val, key) => {
+		if (!isValidShortText(key)) {
+			console.log('debug: Non-valid short text env var key passed to validation.isValidEnv');
+			console.log(`\tKey: ${inspect(key)}`)
+			return false;
+		}
+
+		if (!ENV_VAR_KEY_REGEX.test(key)) {
+			console.log('debug: Invalid env var key passed to validation.isValidEnv');
+			console.log(`\tKey: ${inspect(key)}`)
+			return false;
+		}
+
+		if (!_.isString(val)) {
+			console.log('debug: Non-string value passed to validation.isValidEnv');
+			console.log(`\tval: ${inspect(key)}`);
+			return false;
+		}
+		return true;
+	});
+}
+
+/**
+ * isValidLabelsObject
+ *
+ * Given a labels object, test the types and values for validity
+ */
+export function isValidLabelsObject(obj: LabelObject): boolean {
+	if (!_.isObject(obj)) {
+		console.log('debug: Non-object passed to validation.isValidLabelsObject');
+		console.log(`\tobj: ${inspect(obj)}`);
+		return false;
+	}
+
+	return _.every(obj, (val, key) => {
+		if (!isValidShortText(key)) {
+			console.log('debug: Non-valid short text label key passed to validation.isValidLabelsObject');
+			console.log(`\tkey: ${inspect(key)}`);
+			return false;
+		}
+
+		if (!LABEL_NAME_REGEX.test(key)) {
+			console.log('debug: Invalid label name passed to validation.isValidLabelsObject');
+			console.log(`\tkey: ${inspect(key)}`);
+			return false;
+		}
+
+		if (!_.isString(val)) {
+			console.log('debug: Non-string value passed to validation.isValidLabelsObject');
+			console.log(`\tval: ${inspect(val)}`);
+			return false;
+		}
+
+		return true;
+	});
+}
+
+function undefinedOrValidEnv(val: EnvVarObject): boolean {
+	return !(val != null && !isValidEnv(val));
+}
+
+/**
+ * isValidDependentAppsObject
+ *
+ * Given a dependent apps object from a state endpoint, validate it
+ *
+ * TODO: Type the input
+ */
+export function isValidDependentAppsObject(apps: any): boolean {
+	if (!_.isObject(apps)) {
+		console.log('debug: non-object passed to validation.isValidDependentAppsObject');
+		console.log(`\tapps: ${inspect(apps)}`);
+		return false;
+	}
+
+	return _.every(apps, (val, appId) => {
+		val = _.defaults(_.clone(val), {
+			config: undefined,
+			environment: undefined,
+			commit: undefined,
+			image: undefined,
+		});
+
+		if (!(isValidShortText(appId) && checkInt(appId))) {
```

Comment:
Can you refactor this to `if (!isValidShortText(appId) || !checkInt(appId)) {` ?